beater/mux.go

```diff
@@ -0,0 +1,220 @@
+package beater
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/elastic/apm-server/processor"
+	"github.com/elastic/beats/libbeat/logp"
+
+	"compress/gzip"
+	"compress/zlib"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"crypto/subtle"
+
+	"github.com/elastic/beats/libbeat/monitoring"
+)
+
+type chainable func(processor.Processor, Config, reporter) http.Handler
+
+var (
+	serverMetrics  = monitoring.Default.NewRegistry("apm-server.server")
+	requestCounter = monitoring.NewInt(serverMetrics, "requests.counter")
+	responseValid  = monitoring.NewInt(serverMetrics, "response.valid")
+	responseErrors = monitoring.NewInt(serverMetrics, "response.errors")
+
+	errInvalidToken    = errors.New("invalid token")
+	errPOSTRequestOnly = errors.New("only POST requests are supported")
+
+	handlerMap = map[int]chainable{
+		processor.Backend:  backendHandler,
+		processor.Frontend: frontendHandler,
+		processor.Nop:      nopHandler,
```

Comment:
We should call it something related to healthcheck as long as we don't have other things relying on the no-op stuff. In other words, lets make it genereric when there's a use for that. For now, i suggest renaming to `processor.HealthCheck` and `healthCheckHandler`